### DISCUSSION TITLE
Auto-add Tailwind Prefixes

### DISCUSSION THREAD

**Discussion by swandogg** [tailwindlabs/tailwindcss#4491] [Help]
Hey all,

Is there any way (or any packages anyone knows of) to have tailwind prefixes automatically added to already existing classes? I love the concept of the prefixes, but having to add them to each and every class in a large project can get cumbersome...

**Comment by simonswiss** [comment]
Hey!

Not sure about adding prefixes to _already existing classes_ in your HTML.

That said, to make Tailwind add a prefix to each utility class, here's what you can do: 

In your Tailwind config file, you can use [the `prefix` option](https://tailwindcss.com/docs/configuration#prefix):

Example:

```js
// tailwind.config.js
module.exports = {
  prefix: 'tw-',
}
```

This will generate classes like so:

```html
<div class="tw-text-lg md:tw-text-xl tw-bg-red-500 hover:tw-bg-blue-500">
  <!-- -->
</div>
```

**Reply by swandogg** [reply]
hey Simon,

thanks for the reply! what I meant really is that I love the idea of prefixes and am currently using them, but I'm working on a pretty large project and having to manually add the prefix for each and every tailwind class gets pretty time consuming. So I was curious to see if you or anyone in the community knew of any packages or ways to have a prefix used without having to manually put it in for every class name...

so that I could do something like 
```js
// tailwind.config.js
module.exports = {
  prefix: 'tw-',
}
```

and be able to do this
```js
<div class="text-lg md:text-xl bg-red-500 hover:bg-blue-500">
  <!-- -->
</div>
```

and have either tailwind or some sort of post-processor put those prefixes in for me, so that we don't have to do it each time for each file

**Reply by valgeirb** [reply]
I came here searching for this exact thing, did you ever find a way to not having to manually write out your prefixes?

**Reply by swandogg** [reply]
hey @valgeirb! sadly I never really found a way 😔... there were a few plugins/npm packages out there I found that claimed to automatically insert prefixes at compile time but I never really got them to work... though they seemed to work for other people! [this](https://www.npmjs.com/package/tailwind-classname-prefix-loader) was one of them

**Reply by antoniobrandao** [reply]
@simonswiss hello Simon. The only reason why I'm in need of the prefix is to manage scope in production, as the final app will be loaded on top of another app. So all I need (and I think the users above too) is the prefix to be added in the end at compile time.

It's very cumbersome to have to add the prefix in every place, when all I need is the prefix to be added at the time of compilation.

Do you know a solution for this? (adding the prefix at compile time)
